docs/web-apps/automated-testing/cdp.md

```diff
@@ -0,0 +1,228 @@
+
+:::note
+Selenium has stated in their reference that they will eventually move away from CDP, hence they suggest using the agnostic BiDi API, which abstracts away the implementation details of CDP.
+:::
+
+## What You’ll Need
+
+- A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up)).
+- Your Sauce Labs [Username](https://app.saucelabs.com/user-settings).
+- A working development environment for one of the supported languages: Java, Python, or JavaScript (Node.js).
+
+## Enabling CDP / BiDi
+
+In order to make use of the CDP / BiDi functionality, you have three possibilities:
+
+### 1. Using Selenium
+
+If you're using Selenium, you can enable CDP / BiDi by setting the `devTools` parameter in `sauce:options` to `true`.
+<Tabs
+groupId="lang-ex"
+defaultValue="Python"
+values={[
+{label: 'Java', value: 'Java'},
+{label: 'Python', value: 'Python'},
+{label: 'WebdriverIO', value: 'WebdriverIO'},
+]}>
+
+<TabItem value="Java">
+
+```java
+// Add selenium example
+```
+
+</TabItem>
+<TabItem value="Python">
+
+
+```py
+options = ChromeOptions()
+options.browser_version = 'latest'
+options.platform_name = 'Windows 10'
+
+sauce_options = {'username': os.environ["SAUCE_USERNAME"],
+                 'accessKey': os.environ["SAUCE_ACCESS_KEY"],
+                 'devTools': True,
+                 'name': 'My Selenium CDP Test'}
+
+options.set_capability('sauce:options', sauce_options)
+sauce_url = "https://ondemand.us-west-1.saucelabs.com/wd/hub"
+# Alternatively use EU datacenter
+# sauce_url = "https://ondemand.eu-central-1.saucelabs.com/wd/hub"
+
+driver = webdriver.Remote(command_executor=sauce_url, options=options)
+```
+
+</TabItem>
+<TabItem value="WebdriverIO">
+
+```javascript
+// Add selenium example
+```
+
+</TabItem>
+</Tabs>
+
+
+### 2. Using Playwright (Selenium Grid)
+
+Using Playwright in conjunction with Selenium Grid, you need to provide the Selenium Remote URL and the capabilities as environment variables:
+
+<Tabs
+groupId="dc-url"
+defaultValue="us"
+values={[
+{label: 'United States', value: 'us'},
+{label: 'Europe', value: 'eu'},
+]}>
+
+<TabItem value="us">
+
+```
+SELENIUM_REMOTE_URL="https://{SAUCE_USERNAME}:{SAUCE_ACCESS_KEY}@ondemand.us-west-1.saucelabs.com:443/wd/hub"
+SELENIUM_REMOTE_CAPABILITIES='{"browserName": "chrome", "sauce:options": {"devTools": "true", "name": "My Playwright Selenium Grid CDP Test"}}'
+```
+
+</TabItem>
+<TabItem value="eu">
+
+```
+SELENIUM_REMOTE_URL="https://{SAUCE_USERNAME}:{SAUCE_ACCESS_KEY}@ondemand.eu-central-1.saucelabs.com:443/wd/hub"
+SELENIUM_REMOTE_CAPABILITIES='{"browserName": "chrome", "sauce:options": {"devTools": "true", "name": "My Playwright Selenium Grid CDP Test"}}'
+```
+
+</TabItem>
+</Tabs>
+
+For more information, please refer to the [Playwright Selenium Grid documentation](https://playwright.dev/docs/selenium-grid)
+
+### 3. Using Playwright (WebSocket)
```

Comment:
this options is fairly limited, not even supporting latest browsers, i would omit it for now.